src/debugConfigProvider.ts

```diff
@@ -181,16 +182,37 @@ export class DebugConfigurationProvider implements vscode.DebugConfigurationProv
 
             if (!platformPath) { return null; }
 
-            let debugserverPort = await targetCommand.deviceDebugserver({
-                target: target as Device,
-            });
-            if (!debugserverPort) { return null; }
-
-            dbgConfig.iosDebugserverPort = debugserverPort;
-
             dbgConfig.preRunCommands = (dbgConfig.preRunCommands instanceof Array) ? dbgConfig.preRunCommands : [];
             dbgConfig.preRunCommands.push(`script lldb.target.module[0].SetPlatformFileSpec(lldb.SBFileSpec('${platformPath}'))`);
-            dbgConfig.preRunCommands.push(`process connect connect://127.0.0.1:${debugserverPort}`);
+
+            // iOS 17+ uses Apple LLDB directly to create and attach to the debugserver,
+            // below iOS 17 spawn debugserver with ios-deploy
+            let iosMajorVersion = dbgConfig.iosTarget.version.split('.')[0] as number;
+            if (iosMajorVersion < 17) {
+                let debugserverPort = await targetCommand.deviceDebugserver({
+                    target: target as Device,
+                });
+                if (!debugserverPort) { return null; }
+
+                dbgConfig.iosDebugserverPort = debugserverPort;
+                dbgConfig.preRunCommands.push(`process connect connect://127.0.0.1:${debugserverPort}`);
+            } else {
+                // launch the app via devicectl
+                await _execFile(
+                    'xcrun',
+                    ['devicectl', 'device', 'process', 'launch', '--device', target.udid, '--start-stopped', this.ensureBundleId(dbgConfig)],
+                );
+
+                // attach to the app via LLDB
+                dbgConfig.processCreateCommands = (dbgConfig.postRunCommands instanceof Array) ? dbgConfig.processCreateCommands : [];
+                // LLDB `device` command needs Apple's beta LLDB set as backend for CodeLLDB
+                // e.g. set vscode://settings/lldb.library to /Applications/Xcode-beta.app/Contents/SharedFrameworks/LLDB.framework/Versions/A/LLDB
+                dbgConfig.processCreateCommands.push(`script lldb.debugger.HandleCommand("device select ${target.udid}")`);
+                // using detour with `script` command, as direct evocation causes CodeLLDB to crash at the moment
+                // probably due to a bug in CodeLLDB / unusal exit results from the `device` command
+                let processName = path.basename(platformPath).split('.')[0];
+                dbgConfig.processCreateCommands.push(`script lldb.debugger.HandleCommand("device process attach --name '${processName}'")`);
```

Comment:
How reliable attaching by name? Can the process have different name than binary? Should we try attaching to pid instead of name, if we can get pid from the process launch command above?

Reply:
Not sure, pid might be cleaner. It gets returned as `process.processIdentifier` from `xcrun devicectl device process launch --json-output -`.

Cleanest option imo would be to use `lldb` also to start the process, but I couldn't figure out a way to do this with Apples `lldb device`.